api/auth_middleware.go

```diff
@@ -5,14 +5,24 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/sirupsen/logrus"
 	"github.com/topfreegames/Will.IAM/errors"
+	"github.com/topfreegames/Will.IAM/models"
 	"github.com/topfreegames/Will.IAM/usecases"
 	"github.com/topfreegames/extensions/middleware"
 )
 
 type serviceAccountIDCtxKeyType string
 
 const serviceAccountIDCtxKey = serviceAccountIDCtxKeyType("serviceAccountID")
+const keyPairHeader = "KeyPair"
+const bearerTokenHeader = "Bearer"
+
+type authorizationHeader struct {
```

Comment:
I wonder if this type should be defined here or on its own file 🤔 💭 Any thoughts?

Reply:
I don't think there's a perfect answer for that. What I would do is: if this struct is only used in this file, it's ok. If it's going to be used only by this pkg, another file in this pkg. If it's used in more pkgs, I would put it in models.